distributionviewer/core/static/js/app/components/containers/chart-config-container.js

```diff
@@ -0,0 +1,38 @@
+import React from 'react';
+import { connect } from 'react-redux';
+
+import ChartConfig from '../views/chart-config';
+import * as metricApi from '../../api/metric-api';
+
+
+class ChartConfigContainer extends React.Component {
+  componentDidMount() {
+    metricApi.getMetrics();
+  }
+
+  handleSubmit(e) {
+    var metrics = Array.from(
+      document.querySelectorAll('input[type="checkbox"]:checked'),
+      input => input.value
+    ).join(',');
+
+    if (metrics !== '') {
+      window.location = '/?metrics=' + metrics;
+    }
+  }
+
+  render() {
+    var qmetrics = metricApi.getQueryMetrics(this.props.location.query);
```

Comment:
As far as I can tell (I might be missing something), we could put this into `componentDidMount` and it would still work. It would look something like this:

```javascript
componentDidMount() {
  metricApi.getMetrics();
  this.qmetrics = metricApi.getQueryMetrics(this.props.location.query);
}

// ...

render() {
  return (
    <ChartConfig {...this.props} qmetrics={this.qmetrics} handleSubmit={this.handleSubmit} />
  );
}
```

A small change, but it would allow the component to re-render if it needs to for any reason without re-fetching data over HTTP.